src/lib.rs

```diff
@@ -431,3 +433,89 @@ fn collapse_state_maps(map: &BTreeMap<i64, StateGroupEntry>, state_group: i64) -
 
     state_map
 }
+
+impl Config {
+    pub fn new(
+        db_url: String,
+        output_file: String,
+        room_id: String,
+        max_state_group: String,
+        min_saved_rows: String,
+        transactions: bool,
+        level_sizes: String,
+    ) -> Config {
+        if db_url == "" {
+            panic!("db url is required");
+        }
+
+        let mut output: Option<File> = None;
+        if output_file != "" {
+            output = Some(File::create(output_file).unwrap());
```

Comment:
As per above we should return an error rather than panic (since this isn't an error by the programmer). Also, its better style to use `.except("<what failed>")` and comment why its safe to unwrap.

Reply:
Ah yes, not sure why only that one was unwrap()